crates/asm/stf/src/transition.rs

```diff
@@ -36,18 +37,19 @@ pub fn asm_stf<S: AsmSpec>(pre_state: AnchorState, block: Block) -> AnchorState
     // 2. Filter the relevant transactions
     let all_relevant_transactions = group_txs_by_subprotocol(&block.txdata);
 
+    let mut manager = HandlerRelayer::new();
+
     // 3. LOAD: bring each subprotocol into a HandlerRelayer
     let mut loader_stage = SubprotoLoaderStage::new(&pre_state);
-    S::call_subprotocols(&mut loader_stage);
+    S::call_subprotocols(&mut loader_stage, &mut manager);
 
     // 4. PROCESS: feed each subprotocol its slice of txs
-    let mut process_stage =
-        ProcessStage::new(all_relevant_transactions, loader_stage.into_handler());
-    S::call_subprotocols(&mut process_stage);
+    let mut process_stage = ProcessStage::new(all_relevant_transactions);
+    S::call_subprotocols(&mut process_stage, &mut manager);
 
     // 5. FINISH: let each subprotocol process its buffered interproto messages
-    let mut finish_stage = FinishStage::new(process_stage.into_handler());
-    S::call_subprotocols(&mut finish_stage);
+    let mut finish_stage = FinishStage::new();
+    S::call_subprotocols(&mut finish_stage, &mut manager);
```

Comment:
Ahhh noo!

I meant

```rs
struct FinishStage<'m> {
  manager: &'m mut SubprotoManager,
  // ...
}

// so you can do
let mut manager = ...;

let mut loader_stage = LoaderStage::new(&mut manager);
S::call_subprotocols(...);

let mut process_stage = ProcessStage::new(&mut manager);
S::call_subprotocols(...);

let mut finish_stage = FinishStage::new(&mut manager);
S::call_subprotocols(...);
```

This hides the manager from the spec.  Maybe there's some case where passing the manager in would be useful?  I'm not sure.

And then you could make constructing the final state be done in part by the manager at the end instead of rolling it into a stage?  Not sure if that makes it cleaner or not.